tools/swift-api-digester/ModuleAnalyzerNodes.cpp

```diff
@@ -1602,18 +1616,21 @@ SDKContext::shouldIgnore(Decl *D, const Decl* Parent) const {
       // Typealias should have no impact on ABI.
       if (isa<TypeAliasDecl>(VD))
         return true;
+      // "ABI-public" declarations (i.e. either `public` or `@usableFromInline`)
+      // shouldn't be ignored if we're checking ABI.
+      auto accessScope = VD->getFormalAccessScope(
+        /*useDC=*/nullptr,
+        /*treatUsableFromInlineAsPublic=*/true);
+      if (accessScope.isPublic())
+        return false;
```

Comment:
We shouldn't need this change. The `getAccessLevel()` check down several lines should perform as intended. Also, for ABI checking, we should use `getEffectiveAccess()` instead of `getFormalAccess`

Reply:
I think the `getAccessLevel` check further down will exclude `@usableFromInline internal` declarations.

Reply:
It shouldn't because we are using `getEffectiveAccess()` there. Did you pass down `-abi` flag when invoking the tool?

Reply:
> It shouldn't because we are using `getEffectiveAccess()` there.

My mistake, I didn't realize that. I'll remove this redundant code.

> Did you pass down `-abi` flag when invoking the tool?

Yes, I used the `-abi` flag (see top comment for details).

Reply:
Removing this code will also allow me to restore your `@_alwaysEmitIntoClient` check to its original position.

Reply:
Yeah, that change looks suspicious to me too.